core/src/test/java/org/bitcoinj/core/LockTimeTest.java

```diff
@@ -73,11 +76,32 @@ public void unset() {
 
     @Test(expected = IllegalStateException.class)
     public void blockHeight_mismatch() {
-        LockTime.ofTimestamp(Instant.MAX).blockHeight();
+        LockTime.ofTimestamp(Instant.MAX).getBlockHeight().orElseThrow(IllegalStateException::new);
     }
 
     @Test(expected = IllegalStateException.class)
     public void timestamp_mismatch() {
-        LockTime.ofBlockHeight(1).timestamp();
+        LockTime.ofBlockHeight(1).getTimestamp().orElseThrow(IllegalStateException::new);
+    }
+
+    @Test
+    public void testSubtypes() {
+        LockTime timestamp = LockTime.ofTimestamp(Instant.now());
+        LockTime height = LockTime.ofBlockHeight(100);
+        LockTime unset = LockTime.unset();
+
+        assertTrue(timestamp instanceof TimeLock);
+        assertTrue(height instanceof HeightLock);
+        assertTrue(unset instanceof NoLock);
+
+        if (timestamp instanceof TimeLock) {
+            assertTrue(((TimeLock) timestamp).timestamp().isAfter(Instant.EPOCH));
+        }
+        if (height instanceof HeightLock) {
+            assertTrue(((HeightLock) height).blockHeight() > 0);
+        }
+        if (unset instanceof NoLock) {
+            assertTrue(unset.rawValue() == 0);
+        }
     }
```

Comment:
Maybe use 3 separate tests?

Reply:
I could probably write more than three...

Reply:
I mean this specifc test can literally be split in 3 tests (now two, after NoLock has been removed).